contributing.md

```diff
@@ -0,0 +1,135 @@
+# Contribution
+
+Thanks for contributing to Trunk's default plugins! Read on to learn more.
+
+- [Overview](#overview)
+- [Release process](#releases)
+- [Adding new linters](#linters)
+- [Adding new actions](#actions)
+- [Guidelines](#guidelines)
+- [Docs](https://docs.trunk.io)
+
+## Overview
+
+We use this repository to provide our users with default linters and actions. trunk automatically
+adds the following to users' trunk.yaml:
+
+```yaml
+plugins:
+  sources:
+    - id: trunk
+      uri: https://github.com/trunk-io/plugins
+      ref: v0.0.8
+```
+
+Plugins can also be loaded from local paths, as below:
+
+```yaml
+plugins:
+  sources:
+    - id: trunk
+      local: /path/to/repo/root
+```
+
+Run `trunk run toggle-local` to quickly toggle this setting.
+
+Adding a plugin source lets users run `trunk check enable` or `trunk actions enable` with linters
+and actions defined in that plugin. For more information, see our
+[docs](https://docs.trunk.io/docs/plugins).
+
+If you want to request a new feature or get help with a problem, please see our
+[Features Page](https://features.trunk.io/) or our [Slack](https://slack.trunk.io/).
+
+## Releases
```

Comment:
This section still feels a bit detail-heavy and doesn't feel like it naturally answers a user's question.

Stepping back, think about the journey/flow for someone contributing to plugins:

1. uses a linter `magicwand`
2. discovers `trunk`
3. opens PR to trunk-io/plugins with `magicwand` linter definition
4. wants to use `magicwand` in their own repo

IMO, the purpose of this section is to explain to the user what the process for going from step 3 to step 4 is, so we'd want to explain:

* trunk-io/plugins has a release process that is independent of PRs; that is, we don't live at HEAD but rather pin the version of a plugin (explain that we do this for the same reason we pin linter versions)
  * also explain what `ref` field this is (e.g. "in the above example, we're pointing at v0.0.8 of plugins")
* once a plugins release containing the desired change is created, a user can pick those changes up using `trunk upgrade`

----

In a similar vein, there may be other user flows that you have in mind for someone reading this doc. Remember: always keep your audience in mind when writing.

(related/useful: https://nick.groenen.me/posts/the-4-types-of-technical-documentation/)